src/MagicString.js

```diff
@@ -860,7 +860,12 @@ export default class MagicString {
 		const index = original.indexOf(string);
 
 		if (index !== -1) {
-			this.overwrite(index, index + string.length, replacement);
+			if (typeof replacement === 'function') {
+				replacement = replacement(string);
```

Comment:
Can we align or be a subset of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_function_as_the_replacement?

Also could you update the dts file? Thanks

Reply:
Does the dts need to be updated? The current types seem to be compatible with all cases. 🤔